src/hyperion/external_interaction/ispyb/store_datacollection_in_ispyb.py

```diff
@@ -286,19 +303,37 @@ def _mutate_data_collection_params_for_experiment(
         return params
 
     def _store_scan_data(self, conn: Connector):
-        if not self.data_collection_group_id:
-            self.data_collection_group_id = self._store_data_collection_group_table(
-                conn
-            )
-        data_collection_id = self._store_data_collection_table(
-            conn, self.data_collection_group_id
-        )
-        self.data_collection_id = data_collection_id
-        self._store_position_table(conn, data_collection_id)
+        assert (
+            self.data_collection_group_id
+        ), "Attempted to store scan data without a collection group"
+        assert (
+            self.data_collection_id
+        ), "Attempted to store scan data without a collection"
+        self._store_data_collection_group_table(conn)
+        self._store_data_collection_table(conn, self.data_collection_group_id)
+        self._store_position_table(conn, self.data_collection_id)
 
-        return data_collection_id, self.data_collection_group_id
+        return self.data_collection_id, self.data_collection_group_id
 
     def begin_deposition(self) -> IspybIds:
+        # prevent pyright + black fighting
+        # fmt: off
+        with ispyb.open(self.ISPYB_CONFIG_PATH) as conn:
```

Comment:
Should: Have the `assert conn is not None` like in `update_deposition`

Reply:
Maybe - although TBH I would be surprised if it didn't raise an exception in the actual call to open if there was a problem

Reply:
Yh, it will but by asserting it's not `None` here we also can remove some of the `type: ignore` later